### PR TITLE
Batched load of textual inversions

### PR DESCRIPTION
**Pull request by pdoane**
- Only call resize_token_embeddings once per batch as it is the most expensive operation
- Allow pretrained_model_name_or_path and token to be an optional list
- Remove Dict from type annotation pretrained_model_name_or_path as it was not supported in this function
- Add comment that single files (e.g. .pt/.safetensors) are supported
- Add comment for token parameter
- Convert token override log message from warning to info

### CODE REVIEW

**Comment by pdoane** [comment]
Tested on a directory with 19 embeddings and the bulk load reduced the time from 7.5 seconds to 0.5 seconds.

**Comment by pdoane** [comment]
I reduced the token override message from a warning to info as this is common practice in front ends (e.g. both InvokeAI and A1111 do this).

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -576,16 +585,58 @@ def load_textual_inversion(
...
+
+        if isinstance(token, str):
+            tokens = [token]
+        elif token is None:
+            tokens = [None] * len(pretrained_model_name_or_paths)
+        else:
+            tokens = token
+
+        if len(pretrained_model_name_or_paths) != len(tokens):
+            raise ValueError(
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:601]: nice!

File: src/diffusers/loaders.py [hunk 2]
```diff
@@ -576,16 +585,58 @@ def load_textual_inversion(
...
+                f"You have passed a list of models of length {len(pretrained_model_name_or_paths)}, and list of tokens of length {len(tokens)}"
+                f"Make sure both lists have the same length."
+            )
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:604]: Can we add one more check here that makes sure that there are no duplicates are in the list of tokens?

```suggestion
            )
            
        if len(set(tokens)) < len(tokens):
            raise ValueError(f"You have passed a list of tokens that contains duplicates: {tokens}")
```

File: tests/pipelines/test_pipelines.py [hunk 3]
```diff
@@ -575,6 +575,31 @@ def test_text_inversion_download(self):
...
+                assert pipe._maybe_convert_prompt("<*****>", pipe.tokenizer) == "<*****>"
+
+                token = pipe.tokenizer.convert_tokens_to_ids("<******>")
+                assert token == num_tokens + 9, "Added token must be at spot `num_tokens`"
+                assert pipe.text_encoder.get_input_embeddings().weight[-1].sum().item() == 64
+                assert pipe._maybe_convert_prompt("<******>", pipe.tokenizer) == "<******>"
+
+                prompt = "hey <*****> <******>"
+                out = pipe(prompt, num_inference_steps=1, output_type="numpy").images
+                assert out.shape == (1, 128, 128, 3)
```
Comment (patrickvonplaten) [inline tests/pipelines/test_pipelines.py:601]: Very nice test!

**Review by patrickvonplaten — APPROVED** [review]
Looks great! Good job on making the function compatible with a list of textual inversion vectors. 

PR looks good from my end, I'd maybe just add one more assertion statement to make sure the user doesn't pass duplicated tokens

**Comment by patrickvonplaten** [comment]
Hey @pdoane,

Ah sorry my suggestions now breaks when the tokens are all None, could you maybe change the if statement slightly so that `None` values are filtered before checking?

**Comment by pdoane** [comment]
@patrickvonplaten - sorry for the delay! The PR is now updated with the changes you requested.  There is a CI failure that I think is unrelated in repository consistency.  Let me know if I should investigate more closely.

**Comment by patrickvonplaten** [comment]
Thanks a lot for the PR @pdoane !

**Pull request merged**